apptools/undo/abstract_command.py

```diff
@@ -12,65 +12,4 @@
 # Description: <Enthought undo package component>
 # ------------------------------------------------------------------------------
 
-# Enthought library imports.
-from traits.api import Any, HasTraits, Str, provides
-
-# Local imports.
-from .i_command import ICommand
-
-
-@provides(ICommand)
-class AbstractCommand(HasTraits):
-    """The AbstractCommand class is an abstract base class that implements the
-    ICommand interface.
-    """
-
-    #### 'ICommand' interface #################################################
-
-    # This is the data on which the command operates.
-    data = Any
-
-    # This is the name of the command as it will appear in any GUI element.  It
-    # may include '&' which will be automatically removed whenever it is
-    # inappropriate.
-    name = Str
-
-    ###########################################################################
-    # 'ICommand' interface.
-    ###########################################################################
-
-    def do(self):
-        """This is called by the command stack to do the command and to return
-        any value.  The command must save any state necessary for the 'redo()'
-        and 'undo()' methods to work.  The class's __init__() must also ensure
-        that deep copies of any arguments are made if appropriate.  It is
-        guaranteed that this will only ever be called once and that it will be
-        called before any call to 'redo()' or 'undo()'.
-        """
-
-        raise NotImplementedError
-
-    def merge(self, other):
-        """This is called by the command stack to try and merge another
-        command with this one.  True is returned if the commands were merged.
-        'other' is the command that is about to be executed.  If the commands
-        are merged then 'other' will discarded and not placed on the command
-        stack.  A subsequent undo or redo of this modified command must have
-        the same effect as the two original commands.
-        """
-
-        # By default merges never happen.
-        return False
-
-    def redo(self):
-        """This is called by the command stack to redo the command.  Any
-        returned value will replace the value that the command stack references
-        from the original call to 'do()' or previous call to 'redo()'.
-        """
-
-        raise NotImplementedError
-
-    def undo(self):
-        """ This is called by the command stack to undo the command. """
-
-        raise NotImplementedError
+from pyface.undo.api import AbstractCommand
```

Comment:
For maintaining backward compatibility, I'd think the safest thing to do would be `from pyface.undo.abstract_command import *`, assuming the module is moved to pyface as is. e.g. this breaks `from apptools.undo.abstract_command import ICommand`. We believe no one does that, but it costs little to do the safer approach. 🤞  this does not break anything.

Reply:
@rahulporuri should we make this change before I push ahead with the release?

Reply:
i'd actually prefer breaking such imports now - which will make the eventual transition to `pyface` trivial - and cleaner/better because those imports are antipatterns anyway.